package_locator/locator.py

```diff
@@ -123,4 +125,6 @@ def get_repository_url_and_subdir(ecosystem, package):
     elif ecosystem == CARGO:
         repo_url, subdir = get_cargo_location(package)
 
-    return get_base_repo_url(repo_url), postprocess_subdir(subdir)
+    if subdir != None:
+        return get_base_repo_url(repo_url), postprocess_subdir(subdir)
+    return get_base_repo_url(repo_url), None
```

Comment:
Good catch. This is indeed a bug. I introduced it in the last commit on this file, it seems, but I think I processed my data set before introducing this bug. But we definitely need to fix this, and add a test case for this.

Reply:
On a second thought, the change needs to happen within `postprocess_subdir`. That function requires an input validation check.  I'll do that myself. But thanks a lot for catching the bug.

Reply:
I fixed it.